hathor/transaction/transaction_metadata.py

```diff
@@ -40,17 +41,7 @@ class TransactionMetadata:
     accumulated_weight: float
     score: float
```

Comment:
Yes, but it might be expensive to calculate since it runs in O(n). It is even more expensive to calculate for orphan blocks since the `first_block` metadata cannot be used.

Reply:
Maybe it should be part of another "level" of metadata, then? I mean, we have those in this PR which are static and must be set pre-validation. We have the `score`, which is also static, but must be set post-validation. And then we have the existing other dynamic metadata. Would something along those lines make sense for the future?

Reply:
I guess the calculation cost matters. I mean, it's kind os useless to calculate all the static metadata if the transaction will fail validation before the validators get to consume the static metadata. Make sense?

Reply:
Sure. I just mean we could calculate it (the `score`) after verification, which is just before the consensus runs. But just talking about future changes.

Reply:
Just to be clear, it looks like this discussion does not block this PR, but is a design discussion that could affect future PRs.

I think it might be useful to make a graph of the data that each validation needs and possibly provides. Some validations will need `height`,  some other will need `min_height`, there could be a step to calculate the `score` which the consensus would need. Such graph could help use sort the validations by complexity, so we could have validation stages where cheaper validations come first and the more expensive come later.

Reply:
@jansegre that is a great idea and I also think is the best approach for the future, and indeed it does not block this PR.

Reply:
Ok. Let's create an issue to get this done in the future then.

Reply:
Thanks!